deployer/deployer.js

```diff
@@ -78,7 +88,39 @@ class Deployer {
 		await this._postValidateTransaction(contractCopy, transaction, transactionReceipt);
 
 		const deploymentResult = await this._generateDeploymentResult(contractCopy, transaction, transactionReceipt);
-		await this._logAction(this.constructor.name, contractCopy.contractName, transaction.hash, 0, transaction.gasPrice.toString(), transactionReceipt.gasUsed.toString(), deploymentResult.contractAddress);
+		await this._logAction(this.constructor.name, contractCopy.contractName, transaction.hash, 0, transaction.gasPrice.toString(), transactionReceipt.gasUsed.toString(), deploymentResult.contractAddress, deploymentResult._contract.compiler.version);
+
+		return deploymentResult;
+	}
+
+
+	async deployAndVerify(contract, libraries) {
+		if (!this.defaultOverrides || !this.defaultOverrides.apiKey) {
+			throw new Error('Please provide Etherscan API key!')
+		}
+		const deploymentArguments = Array.prototype.slice.call(arguments);
+		deploymentArguments.splice(0, 2);
+
+		await this._preValidateArguments(contract, deploymentArguments);
```

Comment:
Consider not coping all the functionality from above, but calling the method above, and work with the result